From afe8e1ecc80c3ad20710336fcd9a73eaadf7b2ea Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20Pierret=20=28fepitre=29?=
 <frederic.pierret@qubes-os.org>
Date: Sun, 15 Dec 2019 18:39:15 +0100
Subject: [PATCH] Use installer kernel parameters as default for installed

This way if any kernel parameter was need to boot Qubes on particular hardware, it will also be set to installed system

Fixes QubesOS/qubes-issues#1650

Adapted from a98ba275162937a56f7ca1ee5428b00de883677a
---
 pyanaconda/modules/storage/bootloader/base.py | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/pyanaconda/modules/storage/bootloader/base.py b/pyanaconda/modules/storage/bootloader/base.py
index b8ab5b2a38..d6eae2cc99 100644
--- a/pyanaconda/modules/storage/bootloader/base.py
+++ b/pyanaconda/modules/storage/bootloader/base.py
@@ -842,11 +842,13 @@ class BootLoader(object):
 
     def _preserve_some_boot_args(self):
         """Preserve some of the boot args."""
-        for opt in conf.bootloader.preserved_arguments:
-            if opt not in kernel_arguments:
+        global_no_preserve_args = ["stage2", "root", "rescue", "rd.live.check",
+                                   "ip", "repo", "ks", "rd.md", "rd.luks",
+                                   "rd.dm", "rd.lvm.lv", "rd.lvm", "rd.neednet"]
+        for opt, arg in kernel_arguments.items():
+            if opt in global_no_preserve_args:
                 continue
 
-            arg = kernel_arguments.get(opt)
             new_arg = opt
             if arg:
                 new_arg += "=%s" % arg
-- 
2.31.1

